### PR TITLE
Feat/29 federation privatekey config

### PR DESCRIPTION
**Pull request by caarloshenriq**
This pull request introduces support for a required `-federation_privatekey` argument for Signet chains, ensuring that a valid extended private key is provided and properly handled throughout node initialization. It also adds comprehensive tests and updates functional test scripts to enforce and verify this requirement.

**Signet federation private key support and validation:**

* Added a new `-federation_privatekey` argument, required for Signet chains, which must be a non-empty, valid extended private key (`tprv...`). This argument is registered as sensitive and is validated during startup. (`src/chainparamsbase.cpp`, `src/chainparams.cpp`, `src/chainparams.h`, `src/kernel/chainparams.h`) [[1]](diffhunk://#diff-c534c365a32e58168519cb10607734c54323b7f79049dd9a0a3dfbdfd649b31cR26) [[2]](diffhunk://#diff-ff53e63501a5e89fd650b378c9708274df8ad5d38fcffa6c64be417c4d438b6dR42-R47) [[3]](diffhunk://#diff-95b7d64e534396fc456c9d05b390406fb5b1fcd75996aefe1a0da9369f43d71eR15-R19) [[4]](diffhunk://#diff-652e7e6ccc854c9d38b42e7754bb633a6b9c1f683a38816fb50c0f7e6745510fR127)
* During node initialization, the private key is checked for presence and validity; errors are raised if missing or invalid. The key is then decoded and stored in the `NodeContext` for use in block signing. (`src/init.cpp`, `src/node/context.h`) [[1]](diffhunk://#diff-b1e19192258d83199d8adaa5ac31f067af98f63554bfdd679bd8e8073815e69dR918-R930) [[2]](diffhunk://#diff-b1e19192258d83199d8adaa5ac31f067af98f63554bfdd679bd8e8073815e69dR1376-R1384) [[3]](diffhunk://#diff-e01ee77912a0e7918e5c13b188525ce96597727dab3694095f5bb83d7676edc5R18) [[4]](diffhunk://#diff-e01ee77912a0e7918e5c13b188525ce96597727dab3694095f5bb83d7676edc5R93-R94)

**Testing and validation improvements:**

* Added a new unit test suite (`federation_config_tests.cpp`) to cover argument parsing, validation, and decoding for `-federation_privatekey`, including edge cases (empty, missing, invalid, and valid keys). (`src/test/federation_config_tests.cpp`, `src/test/CMakeLists.txt`) [[1]](diffhunk://#diff-ca35ff6e2c9314303cbda88f093a6533bfdec70dd515999c248d6fbd9abfb026R1-R83) [[2]](diffhunk://#diff-921b2054f6bf380eb08d5c3c21cf8d1c7cfee3736227d611400ae1a13ab3d187R45)
* Updated functional tests for Signet and Quorumeum to include the required federation key argument in node startup, and added checks to ensure node startup fails if the key is missing or invalid. (`test/functional/feature_signet.py`, `test/functional/feature_quorumeum.py`) [[1]](diffhunk://#diff-a2720860bd83c27dc08427498dcc227e092edd116e93c17e2a5e2fea2b70c779R14-R16) [[2]](diffhunk://#diff-a2720860bd83c27dc08427498dcc227e092edd116e93c17e2a5e2fea2b70c779R34-R40) [[3]](diffhunk://#diff-a2720860bd83c27dc08427498dcc227e092edd116e93c17e2a5e2fea2b70c779L108-R119) [[4]](diffhunk://#diff-9d1907e04da01a6b40c7725287ad252f657a114dd085a255ea9f23fa318af380R31-R33) [[5]](diffhunk://#diff-9d1907e04da01a6b40c7725287ad252f657a114dd085a255ea9f23fa318af380L47-R51) [[6]](diffhunk://#diff-9d1907e04da01a6b40c7725287ad252f657a114dd085a255ea9f23fa318af380L133-R137)

**Codebase maintenance:**

* Included necessary header imports for key decoding and handling. (`src/init.cpp`)

These changes collectively enforce the presence and validity of the federation private key for Signet, improving both security and reliability of block signing in test networks.

### CODE REVIEW

**Comment by satsfy** [comment]
@guibressan

File: src/test/federation_config_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+BOOST_AUTO_TEST_CASE(federation_privatekey_is_registered_sensitive)
+{
+    auto flags = gArgs.GetArgFlags("-federation_privatekey");
+    BOOST_REQUIRE(flags.has_value());
+    BOOST_CHECK(*flags & ArgsManager::SENSITIVE);
+}
+
+BOOST_AUTO_TEST_CASE(federation_privatekey_valid_tprv_decodes)
+{
+    const std::string valid_tprv = "tprv8ZgxMBicQKsPctz81GgKmkU9KjupnEJQvgq2u7Dm15H7owsaoiBk2hCPJsVUhDchxcmxWKzxfxjNiKJbfN1Y5HRrtHGDE5FVCw73nLbhxzz";
```
Comment (satsfy) [inline src/test/federation_config_tests.cpp:28]: i hope this is a random key

**Review by satsfy — APPROVED** [review]
ACK 7314da2dba5ae8f0d5746dd50754e3bfd971e4d8
If CI passes.

File: src/init.cpp [hunk 2]
```diff
@@ -1359,6 +1360,23 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     const ArgsManager& args = *Assert(node.args);
     const CChainParams& chainparams = Params();
 
+    // Validate and store federation key for signet block signing
+    if (args.GetChainType() == ChainType::SIGNET) {
+        const auto federation_privatekey = args.GetArg("-federation_privatekey");
+        if (!federation_privatekey || federation_privatekey->empty()) {
+            return InitError(Untranslated("-federation_privatekey is required for signet. "
```
Comment (0xbrito) [inline src/init.cpp:1367]: Does this make the private key required?

Comment (0xbrito) [reply src/init.cpp:1367]: Ok I think I replied myself. I mean, why should it be like that?

**Review by GuiSchet — APPROVED** [review]
ACK
